### PR TITLE
message: atomic races

### PR DESCRIPTION
**Pull request by mreiferson**
there was the possibility of a race since we were separately checking/storing

cc @schmichael

### CODE REVIEW

**Comment by mreiferson** [comment]
RFR @jehiah

File: message.go [hunk 1]
```diff
@@ -65,17 +65,16 @@ func (m *Message) HasResponded() bool {
...
 	}
 	m.Delegate.OnFinish(m)
-	atomic.StoreInt32(&m.responded, 1)
 }
 
 // Touch sends a TOUCH command to the nsqd which
 // sent this message
 func (m *Message) Touch() {
-	if m.HasResponded() {
+	if !atomic.CompareAndSwapInt32(&m.responded, 0, 1) {
```
Comment (jehiah) [inline message.go]: just `LoadInt32` here since we don't want `Touch()` to mark as responded.

Comment (mreiferson) [inline message.go]: ahh good catch

**Comment by mreiferson** [comment]
fixed

**Pull request merged**